### PR TITLE
feat: add JSON-LD transformers

### PR DESCRIPTION
**Pull request by ronjaquensel**
## What this PR changes/adds

Adds a `dsp-transform` module for transforming between EDC model and JSON-LD. Updates the Catalog model to include `Datasets`, `Distributions` and `DataServices`.

Adds the interfaces/classes for `JsonLdTransformer` and `JsonLdTransformerRegistry` to the `json-ld` module.

## Why it does that

JSON-LD support is required for implementation of the dataspace protocol.

## Further notes

Feedback from the closed PR #2620 has been applied.

## Linked Issue(s)

Relates #2474 

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [x] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

**Review by jimmarino — CHANGES_REQUESTED** [review]
See my comment

File: data-protocols/dsp/dsp-transform/src/main/java/org/eclipse/edc/protocol/dsp/transform/transformer/to/JsonObjectToConstraintTransformer.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ */
+public class JsonObjectToConstraintTransformer extends AbstractJsonLdTransformer<JsonObject, Constraint> {
+
+    public JsonObjectToConstraintTransformer() {
+        super(JsonObject.class, Constraint.class);
+    }
+
+    @Override
+    public @Nullable Constraint transform(@NotNull JsonObject object, @NotNull TransformerContext context) {
+        //TODO check for type of constraint (atomic, and, or, ...)
```
Comment (ndr-brt) [inline data-protocols/dsp/dsp-transform/src/main/java/org/eclipse/edc/protocol/dsp/transform/transformer/to/JsonObjectToConstraintTransformer.java:48]: could this todo converted in an issue?

**Review by ndr-brt — APPROVED** [review]
looks good!

File: data-protocols/dsp/dsp-transform/src/main/java/org/eclipse/edc/protocol/dsp/transform/transformer/to/JsonObjectToConstraintTransformer.java (hunk 1 above)
Comment (ronjaquensel) [reply data-protocols/dsp/dsp-transform/src/main/java/org/eclipse/edc/protocol/dsp/transform/transformer/to/JsonObjectToConstraintTransformer.java:48]: created #2671

**Pull request merged**